src/deploymentSlotActions.ts

```diff
@@ -0,0 +1,99 @@
+        super(output);
+        this.steps.push(new SwapStep(this, azureAccount,slot));
+    }
+
+    protected onExecuteError(step: WizardStep, stepIndex: number, error: Error) {
+        if (error instanceof UserCancelledError) {
+            return;
+        }
+        this.writeline(`Failed to swap deployment slots - ${error.message}`);
+    }
+}
+
+class SwapStep extends WizardStep {
+    private _subscription: SubscriptionModels.Subscription;
+    private _sourceSlot;
+    private _targetSlot;
+    
+    get sourceSlot(): DeploymentSlotNode {
+        return this._sourceSlot;
+    }
+    get targetSlot(): DeploymentSlotNode {
+        return this._targetSlot;
+    }
+    set sourceSlot(slot: DeploymentSlotNode) {
+        this._sourceSlot = slot;
+    }
+    set targetSlot(slot: DeploymentSlotNode) {
+        this._targetSlot = slot;
+    }
+
+    constructor(wizard: WizardBase, readonly azureAccount: AzureAccountWrapper, readonly slot: DeploymentSlotNode) {
+        super(wizard, 'Select a slot to swap with');
+        this.sourceSlot = slot;
+    }
+
+    async prompt(): Promise<void> {
+        const deploymentSlots: DeploymentSlotNode[] = await this.slot.parent.getChildren(this.azureAccount);
+        let otherSlots: QuickPickItemWithData<null>[] = [];
+
+        for (let slot of deploymentSlots) {
+            if (this.slot.label !== slot.label) {
+                // Deployment slots must have an unique name
+                const otherSlot: QuickPickItemWithData<null> = {
+                    label: slot.label,
+                    description: '',
+                    detail: 'Destination Slot',
+                    data: null
+                };
+                otherSlots.push(otherSlot);
+            }
+        }
+            
+        const quickPickOptions = { placeHolder: `"${this.sourceSlot.label}" will be swapped with the destination slot.` };
+        const result = await this.showQuickPick(otherSlots, quickPickOptions);
+
+        if (result) {
+            this.targetSlot = deploymentSlots.find((slot) => {
+                return result.label === slot.label;
+            });
+            const credential = this.azureAccount.getCredentialByTenantId(this.slot.subscription.tenantId);
+            const client = new WebSiteManagementClient(credential, this.slot.subscription.subscriptionId);
+    
+            await client.webApps.swapSlotSlot(this.slot.site.resourceGroup, this.slot.site.repositorySiteName, { targetSlot: this.targetSlot.label, preserveVnet: true }, this.sourceSlot.label, {}, 
+                (err, result, request, response) => {
+                if (!err && response.statusCode === 200) {
+                    this.wizard.writeline(`"${this.targetSlot.label}" was swapped with "${this.sourceSlot.label}"`);
+                } else {
+                    this.wizard.writeline(JSON.parse(err.message).Message);

```

Comment:
It looks a little weird that you have .message twice - is that intentional? It's not supposed to be `JSON.parse(err).Message`?

I have no idea what `err` looks like so as long as you've tested it I'm good

Reply:
This was intentional.  Oddly, the error came back as an object with a message property.  That message prop held a JSON string that had a Message property that holds the error message.